### ISSUE TITLE
App letterboxing on modern phones

### ISSUE THREAD

**Issue by Matthewspear** [bitcraze/crazyflie2-ios-client#27]
I think once a few of the current regressions are fixed (#25, #26) it would be worth another release to the App Store, current one is 2 years old.

Recently in the past year newer phones with an updated screen resolution have come out ( X, XR, XS, XS Max). To avoid breaking changes, apps built before this change run in a letterbox:

![img_8259](https://user-images.githubusercontent.com/1668863/51843469-fc3cd100-230a-11e9-8ab0-210d74e9ac8a.PNG)

I've tested it out and all that needs to be done is rebuild the app targeting iOS 12 to support the new screen resolutions.

Happy to help push to the store given the correct details or setup a CD system to build from Master!

**Comment by ataffanel** [comment]
I agree with making a new version. When I test the current version in the new iPhone emulator there is no lettering so I guess just compiling with the new xcode will solve that?

I have everything setup for releasing to the store so that should not be a problem. Though getting a working travis configuration for the project would be nice to make sure commits builds.

**Comment by Matthewspear** [comment]
Yeah just a rebuild will do it.

In the past I’ve used a combination of Bitrise and Fastlane. Allowing for every push to a develop branch triggering testflight builds and to master a full store release.

In terms of the regressions, if I can solve them I hope to start building up some unit tests too.

**Comment by Matthewspear** [comment]
Bitrise also offer accounts for open source projects too - [here](https://blog.bitrise.io/free-developer-plan-features-for-open-source-projects-on-bitrise)

**Comment by ataffanel** [comment]
This all sounds good and the idea could also be applied to the Android app which is nice :).

We usually use master as development branch and tag releases, but I guess these tools can be applied to this workflow as well.

**Comment by Matthewspear** [comment]
All these actions can be triggered manually, but suit a git-flow approach and feature branches.

Yeah familiar with a bit of Android development but I’m sure setup is similar and Fastlane also supports Android.
